#166265918 List Authors

Hadeneekeh opened the pull request:
### What does this PR do?
Have `GET`  request on list authors endpoint

#### Description of Task to be completed?

- Write unit test for the endpoint

- Create the endpoint for list authors functionality

- Document the feature

#### How should this be manually tested?
After cloning the repo

1. Run `npm run migrate`

2. Then run `npm run dev`

3. Now make a `get` request to `localhost:3000/api/v1/authors`

#### Any background context you want to provide?
I am yet to test the doc

#### What are the relevant pivotal tracker stories?
[#166265918](https://www.pivotaltracker.com/story/show/166265918)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/46038827/59027782-d80d7580-8851-11e9-86dc-28d460ef3fcd.png)

#### Questions:
None

Hadeneekeh committed:
feat(list  Authors):
-Implement list authors functionality
[Delivers  #166265918]

emp-daisy commented:
Please fix the failing test

emp-daisy commented on server/controllers/authorController.js:
```diff
@@ -24,6 +24,21 @@ class AuthorController {
+
+  static async listAuthor(req, res) {
+    try {
+      if (!req.headers.authorization) {
```
This can be a reusable middleware as it might be required in other routes. 
Also what if the authorization is invalid?

OvieMudi committed:
feat(book-listing): add route route for listing books

- add new GET route for listing books
- configure controller method for listing and paginating books
- generate Book objects using faker package and seed Books table
- write tests for book endpoints

[#166265864]

OvieMudi committed:
feat(book-listing): include API docs for book-listing

- update swagger docs with API endpoints

[#166265864]

OvieMudi committed:
remove faker "fake-route" from index file

Hadeneekeh committed:
Fix failing test

emp-daisy committed:
Merge pull request #36 from emp-daisy/ft/166265864/paginate-book-listing

#166265864 get all books and paginate book list

Hadeneekeh committed:
Merge branch 'ft/166265918/list-authors' of https://github.com/emp-daisy/HelloBooksLib into ft/166265918/list-authors

Hadeneekeh committed:
Fix failing test

Hadeneekeh committed:
fixx test bug

Hadeneekeh committed:
fixx test bug

Hadeneekeh committed:
feat(list  Authors):
-Add middleware to take care of authentication and fix test bug
[Updates  #166265918]

PelumiAlesh reviewed (approved):
LGTM
